### PR TITLE
feat(http): add header and query validators

### PR DESCRIPTION
**Pull request by pevisscher**

### CODE REVIEW

**Comment by pevisscher** [comment]
Current dependencies on/for this PR:
* main
  * **PR #31** <a href="https://app.graphite.dev/github/pr/skyleague/therefore/31" target="_blank">Graphite</a>  👈
    * **PR #32** <a href="https://app.graphite.dev/github/pr/skyleague/therefore/32" target="_blank">Graphite</a> 

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/skyleague/therefore/31?utm_source=stack-comment).

File: src/lib/primitives/jsonschema/jsonschema.ts
Comment (MickVanDuijn) [inline src/lib/primitives/jsonschema/jsonschema.ts:148]: this feels like another change snuck into this PR

**Review by MickVanDuijn — CHANGES_REQUESTED** [review]
See inline comments

**Review by MickVanDuijn — APPROVED** [review]

**Comment by pevisscher** [comment]
**Graphite Merge Job**

Current status: 🔄 Waiting for checks to pass

This pull request is currently waiting for CI and mergeability checks to pass before being merged.

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/skyleague/therefore/31?utm_source=stack-comment).


Job Reference: 9BzgHsfDernqvMzHCN2o

**Comment by pevisscher** [comment]
**Graphite Merge Job**

Current status: 🔄 Waiting for checks to pass

This pull request is currently waiting for CI and mergeability checks to pass before being merged.

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/skyleague/therefore/31?utm_source=stack-comment).


Job Reference: 4q7XSmmn1AMhDgyvtGlo

**Pull request merged**
